### DISCUSSION TITLE
TwigBundle complaining about missing kernel.secret, even tough it’s not needed in my case

### DISCUSSION THREAD

**Discussion by fabpico** [symfony/symfony#49455] [Q&A]
### Symfony version(s) affected

6.2

### Description

After installing TwigBundle `^6.2`, and injecting dependency `Twig\Environment`, during compilation, following error is thrown

````
You have requested a non-existent parameter "kernel.secret". Did you mean this: "kernel.charset"?
 /home/api/vendor/symfony/dependency-injection/Compiler/DefinitionErrorExceptionPass.php:51
 /home/api/vendor/symfony/dependency-injection/Compiler/AbstractRecursivePass.php:80
 /home/api/vendor/symfony/dependency-injection/Compiler/DefinitionErrorExceptionPass.php:29
 /home/api/vendor/symfony/dependency-injection/Compiler/AbstractRecursivePass.php:44
 /home/api/vendor/symfony/dependency-injection/Compiler/Compiler.php:73
 /home/api/vendor/symfony/dependency-injection/ContainerBuilder.php:721
 /home/api/vendor/symfony/http-kernel/Kernel.php:487
 /home/api/vendor/symfony/http-kernel/Kernel.php:709
 /home/api/vendor/symfony/http-kernel/Kernel.php:122
 /home/api/vendor/symfony/framework-bundle/Test/KernelTestCase.php:73
...
````

I just have a Symfony application as API, and only want to use Twig for email template rendering. I am not going to use Twig for any login form.

So I see no need to force “kernel.secret” to be configured. I already use the environment variable APP_SECRET for a separate JSON Login, and that needed no extra `framework.yaml` file with extra `secret` configuration.

Any chance to remove that validation? Currently I am forced to add an extra configuration which is only for that the compilation will not crash.

### How to reproduce

1. Initial composer.json

````json
{
  "require": {
    "symfony/runtime": "^6.2",
    "symfony/http-kernel": "^6.2",
    "symfony/framework-bundle": "^6.2",
    "symfony/console": "^6.2",
    "doctrine/doctrine-bundle": "^2.8",
    "symfony/orm-pack": "^2.3",
    "doctrine/doctrine-migrations-bundle": "^3.2",
    "symfony/uid": "^6.2",
    "firebase/php-jwt": "^6.3",
    "symfony/mailer": "^6.2",
    "symfony/translation": "^6.2"
  },
  "require-dev": {
    "phpunit/phpunit": "^9.5",
    "symfony/maker-bundle": "^1.48",
    "symfony/security-bundle": "^6.2",
    "phpspec/prophecy": "^1.17"
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "Tests\\": "tests/"
    }
  },
  "config": {
    "allow-plugins": {
      "symfony/runtime": true
    }
  }
}
````

2. Install bundle `composer require symfony/twig-bundle`.

3. bundles.php

````php
<?php declare(strict_types=1);

return [
    Symfony\Bundle\FrameworkBundle\FrameworkBundle::class => ['all' => true],
    Symfony\Bundle\MakerBundle\MakerBundle::class => ['all' => true],
    Doctrine\Bundle\DoctrineBundle\DoctrineBundle::class => ['all' => true],
    Symfony\Bundle\SecurityBundle\SecurityBundle::class => ['all' => true],
    Doctrine\Bundle\MigrationsBundle\DoctrineMigrationsBundle::class => ['all' => true],
    Symfony\Bundle\TwigBundle\TwigBundle::class => ['all' => true], // <-- added this
];
````
4. Inject `Twig\Environment` in any service
5. Make any request to the application

**Comment by MatTheCat** [comment]
`%kernel.secret%` is required by `UriSigner`, required by `FragmentUriGenerator`, required by `HttpKernelRuntime`.

Since the latter is optional, you can remove the `fragment.uri_generator` definition as a workaround.

**Comment by fabpico** [comment]
@MatTheCat In my entire project I have nothing defined about `fragment.uri_generator` or `fragment` or `uri_generator`. Not sure what you mean?

**Reply by MatTheCat** [reply]
`fragment.uri_generator` is unconditionally registered by Symfony: https://github.com/symfony/symfony/blob/57901a1122aeb23b4d71befda2699fda7a0bd9f3/src/Symfony/Bundle/FrameworkBundle/Resources/config/fragment_renderer.php#L36

You can still remove it yourself using a [compiler pass](https://symfony.com/doc/current/service_container/compiler_passes.html).
